sql-plugin/src/main/scala/org/apache/spark/sql/rapids/predicates.scala

```diff
@@ -70,63 +65,52 @@ case class GpuAnd(left: Expression, right: Expression) extends CudfBinaryOperato
 
   override def binaryOp: BinaryOp = BinaryOp.LOGICAL_AND
 
-  import GpuLogicHelper._
-
-  // The CUDF implementation of `and` will return a null if either input is null
-  // Spark does not.
-  // |LHS/RHS| TRUE  | FALSE | NULL  |
-  // |TRUE   | TRUE  | FALSE | NULL  |
-  // |FALSE  | FALSE | FALSE | FALSE |
-  // |NULL   | NULL  | FALSE | NULL  |
-  // So we have to make some adjustments.
-  //  IF (A <=> FALSE, FALSE,
-  //    IF (B <=> FALSE, FALSE,
-  //      A cudf_and B))
-
   override def doColumnar(lhs: GpuColumnVector, rhs: GpuColumnVector): ColumnVector = {
     val l = lhs.getBase
     val r = rhs.getBase
-    withResource(Scalar.fromBool(false)) { falseVal =>
-      val firstPass = withResource(l.and(r)) { lAndR =>
-        withResource(l.equalToNullAware(falseVal)) { lIsFalse =>
-          lIsFalse.ifElse(falseVal, lAndR)
-        }
-      }
-      withResource(firstPass) { firstPass =>
-        withResource(r.equalToNullAware(falseVal)) { rIsFalse =>
-          rIsFalse.ifElse(falseVal, firstPass)
+    withResource(new Table(l, r)) {
+      table => {
+        val astOp = new ast.BinaryOperation(ast.BinaryOperator.NULL_LOGICAL_AND,
+          new ast.ColumnReference(0),
+          new ast.ColumnReference(1))
+        withResource(astOp.compile()) { compiledAstExpr =>
```

Comment:
Do you know how much time is spent in compiling the op? I don't have a good way to cache it and clean up afterwards properly, but I would like to know how much it might be contributing?